azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/repository/MessageRepository.kt

```diff
@@ -22,6 +22,10 @@ private val emptyMessage = MessageInfoModel(
 // - messageDeleted
 internal interface MessageRepositoryMiddlewareInterface {
     fun addLocalMessage(messageInfoModel: MessageInfoModel)
+    fun addPage(page: List<MessageInfoModel>)
+    fun addServerMessage(message: MessageInfoModel)
+    fun removeMessage(message: MessageInfoModel)
+    fun editMessage(message: MessageInfoModel)
 }
 
 internal class MessageRepository : List<MessageInfoModel>, MessageRepositoryMiddlewareInterface {
```

Comment:
Are the List APIs are intentionally tightly bind to MessageRepository?
If not, can we have the APIs public for MessageRepository Interface instances? That way it will be easier to retrieve messages. 

We could simply achieve that by making "MessageRepositoryMiddlewareInterface" implementing the List<object> interface.